python/cloudbuild.yaml

```diff
@@ -13,33 +13,47 @@
 # limitations under the License.
 
 steps:
-  - name: python:3.7
-    id: 'run-unit-tests'
+  - id: 'Unit testing'
+    name: python:3.7-slim-stretch
+    dir: python
     args: ["./run_unit_tests.sh"]
 
-  - name: 'gcr.io/google.com/cloudsdktool/cloud-sdk'
-    id: 'build-dataflow-container'
-    entrypoint: 'gcloud'
-    args: [ 'builds', 'submit', '--project=$PROJECT_ID', '--tag', 'gcr.io/$PROJECT_ID/$_TARGET_GCR_IMAGE']
+  - id: "Build the container image"
+    name: "gcr.io/cloud-builders/docker"
+    args: ["build", "-t", "gcr.io/${PROJECT_ID}/${_IMAGE_NAME}:${_IMAGE_TAG}", python]
+
+  - id: "Push the container image"
+    name: "gcr.io/cloud-builders/docker"
+    args: ["push", "gcr.io/${PROJECT_ID}/${_IMAGE_NAME}:${_IMAGE_TAG}"]
 
-  - name: 'gcr.io/google.com/cloudsdktool/cloud-sdk'
-    id: 'build-dataflow-template'
+  - id: 'Build dataflow template'
+    name: 'gcr.io/google.com/cloudsdktool/cloud-sdk'
     entrypoint: 'gcloud'
-    args: [ 'dataflow', 'flex-template', 'build', '$_TEMPLATE_GCS_LOCATION', '--image', 'gcr.io/$PROJECT_ID/$_TARGET_GCR_IMAGE', '--sdk-language', 'PYTHON', '--metadata-file', 'ml_preproc/spec/metadata.json']
+    args: [ 
+      'dataflow', 
+      'flex-template', 
+      'build',
+      'gs://${_BUCKET_NAME}/dataflow/templates/${_IMAGE_NAME}.json', 
+      '--image=gcr.io/${PROJECT_ID}/${_IMAGE_NAME}:${_IMAGE_TAG}',
+      '--sdk-language=PYTHON', 
+      '--metadata-file=template_spec/metadata.json'
```

Comment:
Why did you move the metadata file outside the Python module? It has a "Setup File" field that is only applicable to the Python pipeline